Blish HUD/GameServices/GameIntegration/AudioIntegration.cs

```diff
@@ -1,38 +1,66 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
 using Blish_HUD.Debug;
 using Blish_HUD.GameServices;
+using Blish_HUD.Settings;
 using CSCore.CoreAudioAPI;
+using Humanizer.DateTimeHumanizeStrategy;
 using Microsoft.Xna.Framework;
+using SharpDX.MediaFoundation;
 
 namespace Blish_HUD.GameIntegration {
     public sealed class AudioIntegration : ServiceModule<GameIntegrationService> {
-
         private static readonly Logger Logger = Logger.GetLogger<AudioIntegration>();
 
-        private const int   CHECK_INTERVAL     = 250;
-        private const int   AUDIOBUFFER_LENGTH = 20;
-        private const float MAX_VOLUME         = 0.4f;
-
-        private readonly MMDeviceEnumerator _deviceEnumerator;
-        private readonly RingBuffer<float>  _audioPeakBuffer = new RingBuffer<float>(AUDIOBUFFER_LENGTH);
+        public enum Devices {
+            Gw2OutputDevice,
+            DefaultDevice
+        }
 
-        private AudioSessionManager2  _activeAudioDeviceManager;
-        private AudioMeterInformation _processMeterInformation;
+        private const string    APPLICATION_SETTINGS = "OverlayConfiguration";
+        private const string    USEGAMEVOLUME_SETTINGS = "GameVolume";
+        private const string    VOLUME_SETTINGS = "Volume";
+        private const string    DEVICE_SETTINGS = "OutputDevice";
+        private const int       CHECK_INTERVAL = 250;
+        private const int       AUDIO_DEVICE_UPDATE_INTERVAL = 5000;
+        private const int       AUDIOBUFFER_LENGTH = 20;
+        private const float     MAX_VOLUME = 0.4f;
+        private readonly        SettingEntry<bool> _useGameVolume;
+        private readonly        SettingEntry<Devices> _deviceSetting;
+        private readonly        MMDeviceEnumerator _deviceEnumerator;
+        private readonly        RingBuffer<float> _audioPeakBuffer = new RingBuffer<float>(AUDIOBUFFER_LENGTH);
+        private readonly        SettingEntry<float> _volumeSetting;
+        
+        private readonly List<(MMDevice AudioDevice, AudioMeterInformation MeterInformation)> _gw2AudioDevices = new List<(MMDevice AudioDevice, AudioMeterInformation MeterInformation)>();
 
         private double _timeSinceCheck = 0;
+        private double _timeSinceAudioDeviceUpdate = 0;
 
-        private float? _averageGameVolume;
+        private float? _volume;
 
         /// <summary>
-        /// Provides an estimated volume level for the application
-        /// based on the volumes levels exhibited by the game.
+        /// This either provides  an estimated volume level for the application
+        /// based on the volumes levels exhibited by the game
+        /// or
+        /// the set volume in settings.
         /// </summary>
-        public float AverageGameVolume => _averageGameVolume ??= CalculateAverageVolume();
+        public float Volume => _volume ??= GetVolume();
```

Comment:
Noting a breaking change here.

Reply:
I don't believe any modules currently rely on this at the moment.  The first one that comes to mind relies on its own master volume multiplier.

https://github.com/OpNop/UndaDaSea-BlishHUD/blob/master/UndaDaSea_BlishHUD.cs

Reply:
Hmmm, yeah, didn't think about it, normally things available to dependencies outside of the main applicaton have an interface. Thought the naming would be more appropriate now but it isn't smth that would break the code if the name would still be `AverageGameVolume`

Reply:
Yes, that's fine.  Was just noting that it's a breaking change.